valkyrie/lib/valkyrie/storage/disk.rb

```diff
@@ -34,6 +34,16 @@ def file_path(id)
     def find_by(id:)
       return unless handles?(id: id)
       Valkyrie::StorageAdapter::File.new(id: Valkyrie::ID.new(id.to_s), io: ::File.open(file_path(id), 'rb'))
+    rescue Errno::ENOENT
+      Valkyrie.logger.warn "File not found on disk for #{id}"
+      nil
```

Comment:
update YARD to indicate it returns nil if not found. or would it be possible/nice to try to use the null object pattern here?

Reply:
I don't think we should return the a null object, because I'm not sure what dealing with an "empty" file would do in a real context. Maybe we should throw an exception like `MetadataAdapter`'s `find_by`?

Reply:
Maybe an exception would be better for now. e.g. here's a place where nil would cause a problem: https://github.com/samvera-labs/valkyrie/blob/309f7c9a9590074b2dd6853afc681ab9d679b8e9/spec/derivative_services/image_derivative_service_spec.rb#L53

Reply:
Yeah, I really don't think a null object is the pattern here. It seems like at some point they'd want to see if the file exists or not, which would be hard if something always comes back.

Reply:
I'll do an exception and run the figgy tests to see how much changes.